src/StaticWebAssetsSdk/Tasks/Compression/ResolveCompressedAssets.cs

```diff
@@ -115,16 +115,22 @@ public override bool Execute()
         // Process the final set of candidate assets, deduplicating assets to be compressed in the same format multiple times and
         // generating new a static web asset definition for each compressed item.
         var formats = SplitPattern(Formats);
-        var assetsToCompress = new List<ITaskItem>();
+        var assetsToCompress = new ITaskItem[matchingCandidateAssets.Count * formats.Length];
```

Comment:
Instead of creating a list and having to turn it into an array, we create an array of the max size we expect to hold and return that directly.

This takes advantage of the fact that MSBuild will ignore any item that is null when reading the outputs from a task, which save us from having to reallocate the list and convert it back into an array of contiguous elements